### PR TITLE
feat(extract): wire connection detection into CLI with performance instrumentation (#239)

### LINKED ISSUE #239: [M1-D1.8/D1.9] Wire connection extraction into CLI with performance instrumentation
### Context
**PRD:** \`docs/project/PRD/active/PRD-phase-12-connection-detection.md\` — M1 D1.8, D1.9

Connection detection must be wired into the existing \`riviere extract\` CLI command. This includes the breaking output format change, new flags (--patterns, --stats), and performance instrumentation showing per-phase timing.

### Traceability
- **PRD Section:** M1-D1.8 (Performance instrumentation), M1-D1.9 (CLI integration)
- **Functional Requirements:** CLI flags, output format change, timing display
- **Success Criteria:** SC-7 (performance baselines documented)

### Acceptance Criteria
- ✓ \`riviere extract\` output changes from \`{ success: true, data: EnrichedComponent[] }\` to \`{ success: true, data: { components: EnrichedComponent[], links: ExtractedLink[] } }\` (breaking change)
- ✓ \`--components-only\` flag continues to output components without links (backward compat)
- ✓ \`--patterns\` flag enables Configurable layer in addition to Golden Path
- ✓ \`--allow-incomplete\` enables lenient mode (emit uncertain links with _uncertain instead of failing)
- ✓ \`--stats\` appends summary: connection counts by type (sync/async), by detection method, uncertain link count
- ✓ \`--dry-run\` runs full extraction, outputs to stdout, does not write file
- ✓ Performance timing: extraction completed summary line with per-phase breakdown (call graph, detection, filtering)
- ✓ When no connections detected, links is empty array
- ✓ Output conforms to Riviere schema Link type
- ✓ Structured errors for connection detection failures use existing CLI error format with new CliErrorCode values

### Edge Case Scenarios
- No components extracted → links is empty array, no connection detection attempted
- Connection detection fails in strict mode → structured error with file/line/reason, non-zero exit
- --components-only with --stats → stats only show component counts (no connection stats)
- --dry-run with --stats → stats displayed to stderr, JSON to stdout
- --patterns without any connection patterns in config → Golden Path only, no error
- --allow-incomplete with no uncertain links → same output as strict mode

### Implementation Guidelines
- **Architecture:** Extend \`features/extract/\` in riviere-cli per §9.1.4
- **Firm constraints:** §9.1.4 (extend existing extract command, not new command)
- **Flexible decisions:** Implementation details of flag wiring
- Wire \`detectConnections(project, enrichedComponents, config)\` from riviere-extract-ts
- Add new CliErrorCode values for connection detection failures (UNRESOLVABLE_TYPE, AMBIGUOUS_INTERFACE, MISSING_EVENT_MATCH)
- Follow existing extract entrypoint pattern in \`packages/riviere-cli/src/features/extract/entrypoint/extract.ts\`
- Performance timing: wrap each phase in timing calls, format as final summary line

### Embedded Reasoning
**Why:** Connection detection is useless without CLI integration. The extract command is the primary user interface.
**What:** Extended extract command with connection detection wiring, new flags, output format change, performance timing.
**How:** After enrichment phase, call detectConnections(). Merge results into output. Add flag handling. Wrap phases in timing. Display summary.

### Testing Strategy
- **Unit:** Flag parsing for new options. Output format with components + links. Stats formatting. Error code mapping.
- **Integration:** Full extract command producing links from test fixtures. --components-only backward compat. --dry-run output behavior.
- **Edge Cases:** Empty components, strict mode failures, flag combinations.

### Dependencies
Depends on #237 #238

### Verification
\`\`\`bash
pnpm nx test riviere-cli -- --testNamePattern "extract"
pnpm nx test riviere-cli -- --coverage
\`\`\`

### PR DESCRIPTION
**Pull request by NTCoding**
Closes #239

### Context
**PRD:** \`docs/project/PRD/active/PRD-phase-12-connection-detection.md\` — M1 D1.8, D1.9

Connection detection must be wired into the existing \`riviere extract\` CLI command. This includes the breaking output format change, new flags (--patterns, --stats), and performance instrumentation showing per-phase timing.

### Traceability
- **PRD Section:** M1-D1.8 (Performance instrumentation), M1-D1.9 (CLI integration)
- **Functional Requirements:** CLI flags, output format change, timing display
- **Success Criteria:** SC-7 (performance baselines documented)

### Acceptance Criteria
- ✓ \`riviere extract\` output changes from \`{ success: true, data: EnrichedComponent[] }\` to \`{ success: true, data: { components: EnrichedComponent[], links: ExtractedLink[] } }\` (breaking change)
- ✓ \`--components-only\` flag continues to output components without links (backward compat)
- ✓ \`--patterns\` flag enables Configurable layer in addition to Golden Path
- ✓ \`--allow-incomplete\` enables lenient mode (emit uncertain links with _uncertain instead of failing)
- ✓ \`--stats\` appends summary: connection counts by type (sync/async), by detection method, uncertain link count
- ✓ \`--dry-run\` runs full extraction, outputs to stdout, does not write file
- ✓ Performance timing: extraction completed summary line with per-phase breakdown (call graph, detection, filtering)
- ✓ When no connections detected, links is empty array
- ✓ Output conforms to Riviere schema Link type
- ✓ Structured errors for connection detection failures use existing CLI error format with new CliErrorCode values

### Edge Case Scenarios
- No components extracted → links is empty array, no connection detection attempted
- Connection detection fails in strict mode → structured error with file/line/reason, non-zero exit
- --components-only with --stats → stats only show component counts (no connection stats)
- --dry-run with --stats → stats displayed to stderr, JSON to stdout
- --patterns without any connection patterns in config → Golden Path only, no error
- --allow-incomplete with no uncertain links → same output as strict mode

### Implementation Guidelines
- **Architecture:** Extend \`features/extract/\` in riviere-cli per §9.1.4
- **Firm constraints:** §9.1.4 (extend existing extract command, not new command)
- **Flexible decisions:** Implementation details of flag wiring
- Wire \`detectConnections(project, enrichedComponents, config)\` from riviere-extract-ts
- Add new CliErrorCode values for connection detection failures (UNRESOLVABLE_TYPE, AMBIGUOUS_INTERFACE, MISSING_EVENT_MATCH)
- Follow existing extract entrypoint pattern in \`packages/riviere-cli/src/features/extract/entrypoint/extract.ts\`
- Performance timing: wrap each phase in timing calls, format as final summary line

### Embedded Reasoning
**Why:** Connection detection is useless without CLI integration. The extract command is the primary user interface.
**What:** Extended extract command with connection detection wiring, new flags, output format change, performance timing.
**How:** After enrichment phase, call detectConnections(). Merge results into output. Add flag handling. Wrap phases in timing. Display summary.

### Testing Strategy
- **Unit:** Flag parsing for new options. Output format with components + links. Stats formatting. Error code mapping.
- **Integration:** Full extract command producing links from test fixtures. --components-only backward compat. --dry-run output behavior.
- **Edge Cases:** Empty components, strict mode failures, flag combinations.

### Dependencies
Depends on #237 #238

### Verification
\`\`\`bash
pnpm nx test riviere-cli -- --testNamePattern "extract"
pnpm nx test riviere-cli -- --coverage
\`\`\`

## Summary by CodeRabbit

* **New Features**
  * Added --stats (prints component/link counts to stderr), --patterns (pattern-based connection detection), and now includes detected links in extract output; timing info is printed after runs.

* **Documentation**
  * CLI reference and docs updated to describe the new extract flags, link output, and timing/statistics behavior.

* **Tests**
  * Expanded test coverage for extraction, connection detection, stats formatting, and CLI flag handling.

### CODE REVIEW

File: packages/riviere-cli/src/features/extract/__fixtures__/extraction-test-fixtures.ts [hunk 1]
```diff
@@ -2,40 +2,75 @@ import {
...
     throw new TestAssertionError('Expected console output but got empty array')
   }
   const parsed: unknown = JSON.parse(firstLine)
-  if (!isExtractionOutput(parsed)) {
-    throw new TestAssertionError('Invalid extraction output')
+  const result = extractionOutputSchema.safeParse(parsed)
+  if (!result.success) {
+    throw new TestAssertionError(`Invalid extraction output: ${result.error.message}`)
+  }
+  return result.data
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-cli/src/features/extract/__fixtures__/extraction-test-fixtures.ts]: ✅ **Fixed**: Fixed in commit 5a5cd62. Both parse functions now use result.error.message consistently.

---
*Generated by Claude Code*
